commit 9b89495e479c5fedbf3f2eca4f1c4e9dd481265e
Author: Vinay Gannevaram <quic_vganneva@quicinc.com>
Date:   Sat Feb 4 19:29:39 2023 +0530

    wifi: nl80211: Allow authentication frames and set keys on NAN interface
    
    Wi-Fi Aware R4 specification defines NAN Pairing which uses PASN handshake
    to authenticate the peer and generate keys. Hence allow to register and transmit
    the PASN authentication frames on NAN interface and set the keys to driver or
    underlying modules on NAN interface.
    
    The driver needs to configure the feature flag NL80211_EXT_FEATURE_SECURE_NAN,
    which also helps userspace modules to know if the driver supports secure NAN.
    
    Signed-off-by: Vinay Gannevaram <quic_vganneva@quicinc.com>
    Link: https://lore.kernel.org/r/1675519179-24174-1-git-send-email-quic_vganneva@quicinc.com
    Signed-off-by: Johannes Berg <johannes.berg@intel.com>

diff --git a/include/uapi/linux/nl80211.h b/include/uapi/linux/nl80211.h
index 831660956ab2..f14621a954e1 100644
--- a/include/uapi/linux/nl80211.h
+++ b/include/uapi/linux/nl80211.h
@@ -6323,6 +6323,9 @@ enum nl80211_feature_flags {
  *
  * @NL80211_EXT_FEATURE_PUNCT: Driver supports preamble puncturing in AP mode.
  *
+ * @NL80211_EXT_FEATURE_SECURE_NAN: Device supports NAN Pairing which enables
+ *	authentication, data encryption and message integrity.
+ *
  * @NUM_NL80211_EXT_FEATURES: number of extended features.
  * @MAX_NL80211_EXT_FEATURES: highest extended feature index.
  */
@@ -6392,6 +6395,7 @@ enum nl80211_ext_feature_index {
 	NL80211_EXT_FEATURE_RADAR_BACKGROUND,
 	NL80211_EXT_FEATURE_POWERED_ADDR_CHANGE,
 	NL80211_EXT_FEATURE_PUNCT,
+	NL80211_EXT_FEATURE_SECURE_NAN,
 
 	/* add new features before the definition below */
 	NUM_NL80211_EXT_FEATURES,